### PR TITLE
refactor(auth-js): session provider

### PR DESCRIPTION
**Pull request by divyam234**
@yusukebe This pr improves Session Provider implementation which was initially taken from next-auth. Composing useEffect logic into separate hooks  to make code more readable and  change in login  function to support Popup Window login which will be added in example repo later.

### CODE REVIEW

File: packages/auth-js/README.md [hunk 1]
```diff
@@ -16,19 +16,27 @@ Before starting using the middleware you must set the following environment vari
...
-AUTH_URL=#optional
+AUTH_URL=https://example.com/api/auth
 ```
 
 ## How to Use
 
 ```ts
-import { Hono, Context } from 'hono'
-import { authHandler, initAuthConfig, verifyAuth, type AuthConfig } from "@hono/auth-js"
+import { Hono} from 'hono'
```
Comment (neolectron) [inline packages/auth-js/README.md:25]: prettier

**Review by yusukebe — APPROVED** [review]
LGTM

**Comment by yusukebe** [comment]
@divyam234 

Thanks! I've fixed the format myself, and it looks good to me. I'll merge this and ship it.

**Comment by divyam234** [comment]
Popup login is added in example repo this will also enable  to use auth.js for cross domain setup.

**Pull request merged**